src/main/java/org/opensearch/security/authtoken/jwt/JwtVendor.java

```diff
@@ -123,6 +130,59 @@ static JsonWebKey createJwkFromSettings(Settings settings) throws Exception {
         }
     }
 
+    private static JsonWebKey getSigningJwk(String algorithm, Settings settings) throws IllegalArgumentException {
+        JsonWebKey jwk = new JsonWebKey();
+        jwk.setPublicKeyUse(PublicKeyUse.SIGN);
+
+        if (AlgorithmUtils.isOctet(algorithm)) {
+            jwk.setKeyType(KeyType.OCTET);
+            String signingKey = Optional.ofNullable(settings.get("signing_key")).orElseThrow(() -> new IllegalArgumentException("Settings for signing key is missing. Please specify at least the option signing_key with a shared secret."));
+            jwk.setProperty("k", signingKey);
+            jwk.setAlgorithm(algorithm);
+        } else if (AlgorithmUtils.isEcDsaSign(algorithm)) {
+            jwk.setKeyType(KeyType.EC);
+            jwk.setAlgorithm(algorithm);
+            jwk.setProperty(EC_CURVE, calcCurve(algorithm));
+            jwk.setProperty(EC_PRIVATE_KEY, Optional.ofNullable(settings.get("ec_private")).orElseThrow(() -> new IllegalArgumentException("Settings for EC private key is missing. Please specify ec_private with required x and y coordinates.")));
```

Comment:
Can you open an issue detailing the documentation required for using these different signature algorithms and perhaps when they are best used?

Reply:
I've put my thoughts on what I think about different signature algorithms in this comment to have it it one place.
TLDR: EC is the hardest to crack but if we take other metrics into account like or how fast algorithm is it's difficult to pick one over the other.
Here's the link: https://github.com/opensearch-project/security/issues/2680#issuecomment-1580679995